tdrs-backend/tdpservice/data_files/util.py

```diff
@@ -3,54 +3,88 @@
 from io import BytesIO
 import xlsxwriter
 import calendar
-from tdpservice.parsers.models import ParserErrorCategoryChoices
+from django.conf import settings
+from django.core.paginator import Paginator
+from django.db import models
+from django.db.models import Count, Q
+from django.utils.translation import gettext_lazy as _
 
 
-def get_xls_serialized_file(data):
-    """Return xls file created from the error."""
+class ParserErrorCategoryChoices(models.TextChoices):
+    """Enum of ParserError error_type."""
+
+    PRE_CHECK = "1", _("File pre-check")
+    FIELD_VALUE = "2", _("Record value invalid")
+    VALUE_CONSISTENCY = "3", _("Record value consistency")
+    CASE_CONSISTENCY = "4", _("Case consistency")
+    SECTION_CONSISTENCY = "5", _("Section consistency")
+    HISTORICAL_CONSISTENCY = "6", _("Historical consistency")
+
+
+def get_prioritized_queryset(parser_errors):
+    """Generate a prioritized queryset of ParserErrors."""
+    PRIORITIZED_CAT2 = (
+        ("FAMILY_AFFILIATION", "CITIZENSHIP_STATUS", "CLOSURE_REASON"),
+    )
+    PRIORITIZED_CAT3 = (
+        ("FAMILY_AFFILIATION", "SSN"),
+        ("FAMILY_AFFILIATION", "CITIZENSHIP_STATUS"),
+        ("AMT_FOOD_STAMP_ASSISTANCE", "AMT_SUB_CC", "CASH_AMOUNT", "CC_AMOUNT", "TRANSP_AMOUNT"),
+        ("FAMILY_AFFILIATION", "SSN", "CITIZENSHIP_STATUS"),
+        ("FAMILY_AFFILIATION", "PARENT_MINOR_CHILD"),
+        ("FAMILY_AFFILIATION", "EDUCATION_LEVEL"),
+        ("FAMILY_AFFILIATION", "WORK_ELIGIBLE_INDICATOR"),
+        ("CITIZENSHIP_STATUS", "WORK_ELIGIBLE_INDICATOR"),
+    )
+
+    # All cat1/4 errors
```

Comment:
we will need to revisit this in a follow-on 🎫. We noted that we want to exclude the following from STT reports but i think it fell out-of-scope for this ticket when we decided to include the summary worksheet. We'll need to figure out how to handle these. Ultimately, OFA **does not** want these cat 4 errors to prevent records from being stored in the db. Something to explore is whether its easiest to relax the validation or just drop these validators. 

![image](https://github.com/user-attachments/assets/ca8f0ba5-998d-465f-809f-c4fe6becfaec)

Reply:
#3342